src/OrchardCore.Modules/OrchardCore.Contents/Liquid/ContentAnchorTag.cs

```diff
@@ -108,12 +108,14 @@ public async ValueTask<Completion> WriteToAsync(List<FilterArgument> argumentsLi
                 else
                 {
                     var metadata = await contentManager.PopulateAspectAsync<ContentItemMetadata>(displayFor);
-
-                    if (metadata.DisplayRouteValues != null && routeValues != null)
+                    if (metadata.DisplayRouteValues != null)
                     {
-                        foreach (var attribute in routeValues)
+                        if (routeValues != null)
```

Comment:
In my PR I just moved the inner condition to the upper one, coz it will be the same. I'm still wondering how this causes the bug

Reply:
Try it with the dev branch and then with this PR, I think you will understand.

Reply:
Sure, but I remembered I changed the condition in 3 or 4 places

Reply:
hmm, I may say the same thing as @hishamco 
What is the logical difference between this and what @hishamco did?

both need to be true to pass.

Reply:
@Skrypt @hishamco 

You are missing the below line 121. Here as before, if `DisplayRouteValues` is not null and the ambient `routeValues` is null the below line 121 is still executed,  which is no more the case in the current dev branch.

    customAttributes["href"] = urlHelper.Action(metadata.DisplayRouteValues["action"] ...);

Reply:
Seems we should apply the fix for other case

editFor, adminFor, createFor and removeFor

Reply:
Yes indeed, will do